includes/islandora_ordered_zip_newspaper_batch.inc

```diff
@@ -0,0 +1,224 @@
+<?php
+
+/**
+ * @file
+ * Implementation of classes from the batch ingest framework for newspapers.
+ */
+
+/**
+ * Batch preprocessor.
+ */
+class IslandoraOrderedNewspaperBatch extends IslandoraNewspaperBatch {
+  /**
+   * Static function to get object class.
+   */
+  protected static function getObjectClass() {
+    return 'IslandoraOrderedNewspaperIssueBatchObject';
+  }
+
+  /**
+   * Generate output analogous to file_scan_directory().
+   */
+  protected function scanZip($target) {
+    // XXX: This is the same as the parent beyond keeping track of the sequence
+    // and extension inside the ZIP for use in the group down the road.
+    $zip = new ZipArchive();
+    $zip_file = drupal_realpath($target);
+    if (($error = $zip->open($zip_file)) !== TRUE) {
+      return FALSE;
+    }
+    $count = $zip->numFiles;
+    $info = array();
+    for ($i = 0; $i < $count; $i++) {
+      $raw_name = $zip->getNameIndex($i, ZipArchive::FL_UNCHANGED);
+      if (isset($this->parameters['zip_encoding']) && trim($this->parameters['zip_encoding']) !== '') {
+        $name = drupal_convert_to_utf8($raw_name, $this->parameters['zip_encoding']);
+      }
+      else {
+        $name = $zip->getNameIndex($i);
+      }
+      $path_info = pathinfo($name);
+      // Avoid creating entries for directories and some weird Mac stuff...
+      // XXX: Should be made to take the pattern from
+      // $this::getPattern(), and to filter the results based on it.
+      if (isset($path_info['extension']) && strpos($path_info['dirname'], '__MACOSX') === FALSE && $path_info['extension'] != 'DS_Store') {
```

Comment:
Not sure if we care or not, but [it looks like allow `Thumbs.db` through here](http://puu.sh/q2HsO/ef1d51e915.png)... Should we filter 'em out?... Or, GIGO?